### PR TITLE
Make OrbitQt a library target

### PR DESCRIPTION
**Pull request by beckerhe**
Previously OrbitQt was an executable target (Orbit.exe) which makes
problems for tests as these tests can't just link against OrbitQt.

So this commit converts OrbitQt into a library target and moves the
main.cpp into a separate executable target called "Orbit".

Hence, the target OrbitQtTests can just link against OrbitQt.

Test: Ran unit tests.

### CODE REVIEW

**Review by vickyliu-go4it — APPROVED** [review]
Thanks for resolving this. LGTM!

File: src/OrbitQt/FilterPanelWidget.h [hunk 1]
```diff
@@ -8,8 +8,6 @@
 #include <QFrame>
 #include <QWidget>
 
-#include "ui_FilterPanelWidget.h"
-
 namespace Ui {
 class FilterPanelWidget;
 }
```
Comment (antonrohr) [inline src/OrbitQt/FilterPanelWidget.h:13]: Just to clarify. The best practice here is to include `ui_FilterPanelWidget.h` in FilterPanelWidget.cpp and in FilterPanelWidget.h this forward declaration (i hope this is the right term) aka
```
namespace Ui {
class FilterPanelWidget;
}
```
?

**Review by antonrohr — APPROVED** [review]
Thank you very very much! Especially since this blocked me on the target label stuff

File: src/OrbitQt/FilterPanelWidget.h (hunk 1 above)
Comment (beckerhe) [reply src/OrbitQt/FilterPanelWidget.h:13]: Yes, exactly. (Forward declaration is the right term)

**Comment by beckerhe** [comment]
If the CI is happy, I will merge! Thank you both for the review!

**Pull request merged**